src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -248,7 +248,21 @@ private class TraceBinder : SerializationBinder
 
             public override System.Type BindToType(string assemblyName, string typeName)
             {
+                //TODO cache
                 var assemblies = AppDomain.CurrentDomain.GetAssemblies().Where(a => !a.IsDynamic);
+                var assemblyNameObj = new AssemblyName(assemblyName);
+                //find matching assemblies by name, version is not used.
+                var matchingAssembly = assemblies.FirstOrDefault(x => x.GetName().Name == assemblyNameObj.Name);
+                if(matchingAssembly!= null)
+                {
+                   var matchingType = matchingAssembly.GetType(typeName);
+                    if(matchingType != null)
+                    {
+                        return matchingType;
+                    }
+                }
+                //if there was no matching assembly, or type, try all assemblies and all types.
```

Comment:
As discussed add a "TODO" here to revisit removing this code at a later stage. It might not be needed at all.